meal_sharing-project/src/components/MealsList.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from "react";
+
+function MealsList() {
+  const [meals, setMeals] = useState([]);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetch("http://127.0.0.1:5001/api/meals")
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error("Network response was not ok");
+        }
+        return res.text(); // Повертає дані у форматі тексту
+      })
+      .then((data) => {
+        const parsedData = JSON.parse(data);
```

Comment:
The result of `fetch` has an async method named `json` which can be called. [You can read more about that here.](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) 🙂 

It might be beneficial to move this logic into a async function to avoid all the `.then` and callbacks. This could look something like this:
```js
useEffect(() => {
  async function fetchMeals() {
    try {
      const res = await fetch("http://127.0.0.1:5001/api/meals");
    
      if (!res.ok) {
        throw new Error("Network response was not ok");
      }
    
      const parsedData = await res.json();
      setMeals(parsedData);
      setIsLoading(false);
    } catch (error) {
      setError(error);
      setIsLoading(false);
      console.error("There was an error fetching the data:", error);
    }
  }
  fetchMeals();
}, []);
```